### DISCUSSION TITLE
How to get time for each sub-step in BERTopic such as embedding, dimensionality reduction, and so on?

### DISCUSSION THREAD

**Discussion by MRaza98** [MaartenGr/BERTopic#2193] [Q&A] [answered]
I am playing around with some parameters such as reducing number of topics, changing around PCA and UMAP, and within that changing the number of components (dimensions). My dataset is hundreds of thousands of rows of speeches from a parliament so processing takes hours. I would like to understand what step within BERTopic takes how much time.

**Comment by MaartenGr** [comment, marked as answer]
You can't know that beforehand since it depends on many things (dataset size, parameters, CPU, RAM, dataset structure and content, etc.). However, you can get a feeling of the time by setting `verbose=True` as that will show you how long each step has taken.
